### PR TITLE
allow analyzer version 3.0.0

### PR DESCRIPTION
**Pull request by jakemac53**
~Overriding to that version for testing purposes, will remove the override before submitting.~ reverted now, tests were good

Towards https://github.com/dart-lang/dart_style/issues/1078

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -5,15 +5,14 @@ description: >-
   Opinionated, automatic Dart source code formatter.
   Provides an API and a CLI tool.
 repository: https://github.com/dart-lang/dart_style
-
 environment:
-  sdk: '>=2.12.0-0 <3.0.0'
+  sdk: ">=2.12.0-0 <3.0.0"
```
Comment (jakemac53) [inline pubspec.yaml:9]: Not sure where these double quotes came from.... I can revert if you want?

**Review by kevmoo — APPROVED** [review]

File: pubspec.yaml (hunk 1 above)
Comment (kevmoo) [reply pubspec.yaml:9]: Drop the `-0` while you're here!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # 2.2.1-dev
```
Comment (kevmoo) [inline CHANGELOG.md:1]: Are we ready to drop the `-dev` and ship this?

Comment (jakemac53) [reply CHANGELOG.md:1]: Happy to if @munificent would like me to do it in this pr

Comment (kevmoo) [reply CHANGELOG.md:1]: @munificent !!! 😡

File: pubspec.yaml (hunk 1 above)
Comment (jakemac53) [reply pubspec.yaml:9]: 🤷‍♂️ I would really like to avoid changing more than what this PR is really about?

**Pull request merged**